CMakeLists.txt

```diff
@@ -0,0 +1,89 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+
+cmake_minimum_required(VERSION 3.10)
+
+project(CLog)
+
+set(CLOG_ARTIFACT_DIR ${PROJECT_BINARY_DIR}/artifacts)
+
+function(DOT_NET_BUILD)
+    if (WIN32)
```

Comment:
This if needs to be fixed to support things like arm and 32 bit windows, arm linux, and macos x64. I think the .NET team might have a way to get the runtime from the current platform, if not one needs to be made.

Reply:
Actually, since this already depends on .NET SDK being installed, I _think_ that you can just not pass a runtime, and it will still work. It still generates the exe, so find_program works, but it just publishes for the current platform, which is what we need.

Reply:
Yeah, I just tested this, and on windows it works fine. I'll try linux soon. Just remove the entire DOTNET_RUNTIME if, and on line 41 remove --self-contained and -r ${DOTNET_RUNTIME}. Since we're guaranteed to have a .NET SDK installed since we're doing a live build, we don't need a self contained folder for everything to work. And that solves all the issues with all platforms, without having to write anything custom.

Reply:
https://github.com/microsoft/CLOG/pull/87

I just PR'd the fix, along with getting the formatting working again.